app/src/main/java/com/example/drawingactivity/CustomColorViewModel.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.drawingactivity
+
+import android.graphics.Color
+
+/**
+ * A viewModel that controls the colors set in multiple views
+ * primaryColor sets the primary color for the app
+ * ViewStates is used in the lambda onUpdate to update views in whatever activity it is invoked from
+ * cyclePrimaryColor is the internal function that sets a new Primary Color by selecting from a list
+ */
+class CustomColorViewModel(
+
+) {
+    var onUpdate: ((ViewStates) -> Unit)? = null
+
+    data class ViewStates(
+        var primaryColor: Int,
+    )
+
+    private var primaryColor = Color.BLACK
```

Comment:
This could end up in a case where primaryColor has a color not in the list, why not declare the list first and then make this `primaryColors[0]`

Reply:
adjusted to use index 0 in primary colors now, thank you for mentioning that